test/testcafe/spec/SelectAuthenticatorForEnroll_spec.js

```diff
@@ -62,87 +67,140 @@ test.requestHooks(mockEnrollAuthenticatorPassword)('should load select authentic
+  const selectFactorPage = await setup(t);
+  await t.expect(selectFactorPage.getFormTitle()).eql('Set up security methods');
+  await t.expect(selectFactorPage.getFormSubtitle()).eql(
+    'Security methods help protect your account by ensuring only you have access.');
+  await t.expect(selectFactorPage.getFactorsCount()).eql(4);
+
+  await t.expect(selectFactorPage.getFactorLabelByIndex(0)).eql('Password');
+  await t.expect(selectFactorPage.getFactorIconClassByIndex(0)).contains('mfa-okta-password');
+  await t.expect(selectFactorPage.getFactorSelectButtonByIndex(0)).eql('Set up');
+  await t.expect(selectFactorPage.getFactorSelectButtonDataSeByIndex(0)).eql('okta_password');
+  await t.expect(selectFactorPage.getFactorDescriptionByIndex(0)).eql('Choose a password for your account');
+  await t.expect(await selectFactorPage.factorUsageTextExistsByIndex(0)).eql(true);
+  await t.expect(selectFactorPage.getFactorUsageTextByIndex(0)).eql('Used for access');
+
+  await t.expect(selectFactorPage.getFactorLabelByIndex(1)).eql('Phone');
+  await t.expect(selectFactorPage.getFactorIconClassByIndex(1)).contains('mfa-okta-phone');
+  await t.expect(selectFactorPage.getFactorSelectButtonByIndex(1)).eql('Set up');
+  await t.expect(selectFactorPage.getFactorSelectButtonDataSeByIndex(1)).eql('phone_number');
+  await t.expect(selectFactorPage.getFactorDescriptionByIndex(1)).eql('Verify with a code sent to your phone');
+  await t.expect(await selectFactorPage.factorUsageTextExistsByIndex(1)).eql(true);
+  await t.expect(selectFactorPage.getFactorUsageTextByIndex(1)).eql('Used for recovery');
+
+  await t.expect(selectFactorPage.getFactorLabelByIndex(2)).eql('Okta Verify');
+  await t.expect(selectFactorPage.getFactorIconClassByIndex(2)).contains('mfa-okta-verify');
+  await t.expect(selectFactorPage.getFactorSelectButtonByIndex(2)).eql('Set up');
+  await t.expect(selectFactorPage.getFactorSelectButtonDataSeByIndex(2)).eql('okta_verify');
+  await t.expect(selectFactorPage.getFactorDescriptionByIndex(2))
+    .eql('Okta Verify is an authenticator app, installed on your phone, used to prove your identity');
+  await t.expect(await selectFactorPage.factorUsageTextExistsByIndex(2)).eql(true);
+  await t.expect(selectFactorPage.getFactorUsageTextByIndex(2)).eql('Used for access and recovery');
+
+  await t.expect(selectFactorPage.getFactorLabelByIndex(3)).eql('Security Question');
+  await t.expect(selectFactorPage.getFactorIconClassByIndex(3)).contains('mfa-okta-security-question');
+  await t.expect(selectFactorPage.getFactorSelectButtonByIndex(3)).eql('Set up');
+  await t.expect(selectFactorPage.getFactorSelectButtonDataSeByIndex(3)).eql('security_question');
+  await t.expect(selectFactorPage.getFactorDescriptionByIndex(3)).eql('Choose a security question and answer that will be used for signing in');
```

Comment:
also check for `getFactorUsageTextByIndex` ?

Reply:
Security Question authenticator has 'allowedFor' set to 'none' in the remediation form. If 'allowedFor' has anything other than 'sso' or 'recovery' or 'any' then the `.authenticator-usage-text` element itself won't exist. so not checking for its text content here.

Reply:
so lets check for that element doesn't exist.. This can cover a not-so-happy path

Reply:
its being checked at line line #201